q2_fmt/tests/test_engraftment.py

```diff
@@ -909,3 +915,125 @@ def test_sample_id_match(self):
+                    float("Nan")],
+            'subject': [1, 1, 1, 2, float("Nan"),
+                        float("Nan")],
+            'group': [1, 2, 3, 2, float("Nan"),
+                      float("Nan")]}).set_index('id')
+        metadata_obj = Metadata(metadata_df)
+        column_properties = metadata_obj.columns
+        with self.assertRaisesRegex(AssertionError, ".*Column with numeric"
+                                    " values that was selected: `subject`"):
+            _check_subject_column_categorical(column_properties, 'subject')
+
+    def test_reference_column_numeric(self):
+        metadata_df = pd.DataFrame({
+            'id': ['sample1', 'sample2', 'sample3', 'sample4',
+                   "1", "2"],
+            'Ref': [1, 1, 1, 2, float("Nan"),
+                    float("Nan")],
+            'subject': ['sub1', 'sub1', 'sub1', 'sub2', float("Nan"),
+                        float("Nan")],
+            'group': [1, 2, 3, 2, float("Nan"),
+                      float("Nan")]}).set_index('id')
+        metadata_obj = Metadata(metadata_df)
+        column_properties = metadata_obj.columns
+        with self.assertRaisesRegex(AssertionError, ".*Column with numeric"
+                                    " values that was selected: `Ref`"):
+            _check_reference_column_categorical(column_properties, 'Ref')
+
+    def test_reference_series_not_in_table(self):
+        metadata_df = pd.DataFrame({
+            'id': ['sample1', 'sample2', 'sample3', 'sample4',
+                   'donor1', 'donor2'],
+            'Ref': ['1', '1', '2', '2', float("Nan"),
+                    float("Nan")],
+            'subject': ['sub1', 'sub1', 'sub2', 'sub2', float("Nan"),
+                        float("Nan")],
+            'group': [1, 2, 1, 2, float("Nan"),
+                      float("Nan")]}).set_index('id')
+        reference_series = metadata_df['Ref'].dropna()
+        table_df = pd.DataFrame({
+            'id': ['sample1', 'sample2', 'sample3', 'sample4',
+                   'donor1', 'donor2'],
+            'Feature1': [1, 0, 1, 1, 1, 1],
+            'Feature2': [1, 1, 1, 1, 1, 1],
+            'Feature3': [0, 0, 1, 1, 1, 1]}).set_index('id')
+        peds_df = pd.DataFrame(columns=['id', 'measure',
+                                        'transfered_donor_features',
+                                        'total_donor_features', 'donor',
+                                        'subject', 'group'])
+        with self.assertRaisesRegex(AssertionError, ".*['1' '2'].*"):
+            _compute_peds(peds_df=peds_df, peds_type="Sample",
+                          peds_time=np.nan,
+                          reference_series=reference_series,
+                          table=table_df, metadata=metadata_df,
+                          time_column="group", reference_column="Ref",
+                          subject_column="subject")
+
+    def test_subject_column_name_is_ID(self):
```

Comment:
Similar to last comment, if you refactor the subject/reference column check in metadata to be a single helper method, you can pair this down to just one test instead of two.